packages/ui-components/src/Input/TextTypeInput.tsx

```diff
@@ -3,13 +3,14 @@ import classNames from "classnames/bind";
 import { CommonInputProps, CommonInput } from "./CommonInput";
 import { isEmpty } from "lodash";
 import "./input.module.scss";
+import { FieldRenderProps } from "react-final-form";
 
-export interface TextAndNumberProps
-  extends React.InputHTMLAttributes<HTMLInputElement> {
+// extending the fieldrender props here ensures that we have access to meta and input
+// if we need thems
+export interface TextAndNumberProps<T>
+  extends Partial<FieldRenderProps<T, HTMLInputElement>> {
```

Comment:
Partial because we don't want the field props (input, meta) to be mandatory props for the Text input.

Reply:
also, `T` is used here as we need to specify whether a `text/number` input is being used (this type refers to `FieldValue`, which should be specified instead of having a `text | number` type